mlir/test/Ion/PulseOp.mlir

```diff
@@ -0,0 +1,71 @@
+func.func @example_pulse(%arg0: f64) -> !quantum.bit {
+    %0 = quantum.alloc( 1) : !quantum.reg
+    %1 = quantum.extract %0[0] : !quantum.reg -> !quantum.bit
+    ion.pulse(%arg0: f64) %1 {beam=#ion.beam<
+        transition = #ion.transition<level_0 = #ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, level_1 =#ion.level<principal=1, spin=1.1, orbital=2.2, nuclear=3.3, spin_orbital=4.4, spin_orbital_nuclear=5.5, spin_orbital_nuclear_magnetization=6.6, energy=8.8>, einstein_a=10.10>,
+        rabi=10.10,
```

Comment:
Since units were explicitly specified in frontend toml, having an explicit string attribute for units in the mlir (as opposed to keeping it as just another item in the attr-dict) might also be a good idea.

Reply:
My general preference for values representing physical quantities is to make it explicit, either in the documentation or in the variable name itself, that the program assumes the value has specific units, and not try to track any units specified by the user.

For example, the documentation for the `rabi` variables might read,

```
rabi (float): The Rabi frequency in units of [Hz].
```

There might also be a case to go one step further and refactor the variable names with the unit as a suffix, e.g. `rabi_Hz`, although this is more verbose and probably not necessary as long as we're using standard SI units.

While we do have entries for units in the TOML configs, the intention there was more for documentation than information required by the compiler. I'm not sure what we'd gain by tracking the units in the MLIR as a string attribute. If anything it might give the false impression that a user could specify equivalent quantities, e.g. "1000 Hz" = "1 kHz", but this would be a big undertaking to support correctly (assuming there is no MLIR/llvm library that handles this already).

Sound reasonable? Happy to discuss more.